### ISSUE TITLE
Unable to build macOS app with app links in entitlements

### ISSUE THREAD

**Issue by martin-robert-fink** [joaopmarquesini/uni_links_macos#1]
When I add my domain to `DebugProfile.entitlements` as follows:
```
<dict>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>auth.dewne.com</string>
	</array>
... other keys
</dict>
```

My build fails with:
```
Launching lib/main.dart on macOS in debug mode...
lib/main.dart:1
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
error: Provisioning profile "Mac Team Provisioning Profile: com.example.app" doesn't include the currently selected device "Martin’s MacBook Pro M1" (identifier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXXX). (in target 'Runner' from project 'Runner')
** BUILD FAILED **

Exception: Build process failed
Exited (sigterm)
```

I have tried adding the Associated Domains both via Xcode and editing the plist files in VSCode.  When I follow the exact same steps for iOS (which will use uni_links), everything works and builds fine.  Also, the MacOS app builds fine using Xcode - it fails with the above error when building from VSCode.

Here is Flutter doctor:

```
mfink@MacBook-Pro dewne % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, 2.1.0-10.0.pre, on macOS 11.2.2 20D80 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.54.1)
[✓] Connected device (2 available)
```

Since this only happens with the uni_links_macos plugin, I can't quite tell for sure if this is an issue with Dart/Flutter or with this plugin.  Oh, I should mention that the uni_links_macos plugin works fine when using URL Schemes rather than app links (and doesn't require the entitlements).

As an aside, there is an issue with the readme file since most of it was copy/pasted from the iOS/Android version of uni_links.  The plist file for iOS is `Runner.entitlements`, but for MacOS it's `DebugProfile.entitlements` and `Release.entitlements`.

**Comment by joaopmarquesini** [comment]
I'm not sure what causes that as I've never used Associated Domains on macos, but I'm going to take a look.
If you already have any idea on how to fix, please feel free to send a PR.

> As an aside, there is an issue with the readme file since most of it was copy/pasted from the iOS/Android version of uni_links. The plist file for iOS is Runner.entitlements, but for MacOS it's DebugProfile.entitlements and Release.entitlements

Yeah, this is basically a workaround I made to be able to use url_schemes on my macos app. I have no plans of mantaining it after the original uni_links implements macos support in their own version.

**Comment by martin-robert-fink** [comment]
I was using URL Schemes and I'm trying to move to Universal/App Links.  I'm trying to get rid of the user prompts when the redirect occurs from a URL Scheme.

I don't know what's causing this.  It's very odd because it all builds and works from Xcode, but the build fails from VSCode.  I also posted a question on Stack Overflow since it feels to me like it's a flutter issue.

Thanks for looking into it.

Martin

**Comment by martin-robert-fink** [comment]
I posted an issue in Flutter because of the fact that it builds using Xcode, but not with flutter.  See here:

https://github.com/flutter/flutter/issues/77941

**Comment by martin-robert-fink** [comment]
Hi there -

I got a response on the Flutter group.  They couldn't reproduce the issue.  It gave me the idea to try again using my x86 Mac (I have both x86 and M1).  Everything worked on x86.  So, it appears this issue is an M1 Mac specific issue.

Martin

**Comment by joaopmarquesini** [comment]
hmmm good to know.
When you tried to build using URL Schemes and it worked, were you on the M1 or on the x86?
Can you check if the build works without the Entitlements on the M1?

**Comment by martin-robert-fink** [comment]
Yes. I can confirm that all worked well on M1 when using URL Schemes.
